### DISCUSSION TITLE
Integrate into existing project

### DISCUSSION THREAD

**Discussion by mvoss96** [HomeSpan/HomeSpan#909] [Ideas]
Hi, thanks for your amazing project.
I have one suggestion for future Versions.

I have multiple existing Projects (Smart Lights or sensor hubs) where i would like to integrate homekit support however i cant use homespan for these projects because i already am using the serial monitor and also already have my own wifi management and OTA support.

It would be great if there was a way to use homespan just for homekit connection without the additional features.

**Comment by HomeSpan** [comment]
Thanks - making the system more module is something I'll be considering for future versions in conjunction with the implementation of Ethernet support under the new Arduino-ESP32 3.X.
